[bug?(zenn-cli)] throw `Cannot find module 'node:process'` on node v14.17.6

# 起きている事象
zenn-cliをlatestにアップグレードし `zenn preview` を実行すると、以下のエラーが発生します。

```bash
❯ yarn dev            
yarn run v1.22.19
$ zenn preview
internal/modules/cjs/loader.js:892
  throw err;
  ^

Error: Cannot find module 'node:process'
Require stack:
- /app/node_modules/zenn-cli/dist/server/zenn.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
    at Function.Module._load (internal/modules/cjs/loader.js:745:27)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at /app/node_modules/zenn-cli/dist/server/zenn.js:3:2223938
    at /app/text-output-content/node_modules/zenn-cli/dist/server/zenn.js:3:2310989
    at Object.<anonymous> (/app/node_modules/zenn-cli/dist/server/zenn.js:3:2310993)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/node_modules/zenn-cli/dist/server/zenn.js'
  ]
}
```

previewだけでなく、 `new:article` 等のCLIのコマンドも同様のエラーで終了します。
アップグレード前のバージョンは0.1.98, アップグレード後は0.1.113です。
https://github.com/MH4GF/text-output-content/commit/b46cc05631c87b9d0422040d5d8002bf316e2e8d

# Workaround

この問題はnodeのバージョンがv16.13.0以降、v14であればv14.18.0以降であれば発生しません。
ref: https://stackoverflow.com/questions/71473345/whats-the-difference-between-nodeprocess-and-process

まだ詳しく調べきれていないのですが、zenn-cliのソースコードを軽く見たかぎりでは `node:<package_name>` でimportしている箇所は見つけられなかったため、なぜこのエラーが出るのだろう…？となっています。
0.1.98では動作するため、その間のdiffにリグレッションがあるのかもしれません。

もしくは、package.jsonで指定しているenginesを引き上げる解決策もあるかもしれません。
https://github.com/zenn-dev/zenn-editor/blob/971e649d1e340e30c0f4eb0b2e9b195ee28ca9da/packages/zenn-cli/package.json?plain=1#L102

--- a/packages/zenn-cli/package.json
+++ b/packages/zenn-cli/package.json
@@ -99,7 +99,7 @@
     "zenn-markdown-html": "^0.1.114-alpha.0"
   },
   "engines": {
-    "node": ">=10.13.0"
+    "node": ">=14.0.0"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/zenn-cli/webpack.server.js
+++ b/packages/zenn-cli/webpack.server.js
@@ -6,7 +6,7 @@ const webpack = require('webpack');
 module.exports = {
   target: 'node',
 
-  mode: 'development',
+  mode: 'production',
 
   entry: './src/server/zenn.ts',
 

--- a/packages/zenn-cli/webpack.server.js
+++ b/packages/zenn-cli/webpack.server.js
@@ -6,7 +6,7 @@ const webpack = require('webpack');
 module.exports = {
   target: 'node',
 
-  mode: 'production',
+  mode: 'development',
 
   entry: './src/server/zenn.ts',
 
@@ -19,10 +19,25 @@ module.exports = {
   externals: [
     // package.json はビルドファイルには含めず外部ファイルとして読み込む
     // パスはビルド後のファイル構造を考慮する
-    ({ request }, callback) =>
-      /package\.json$/.test(request)
-        ? callback(null, 'commonjs ../../package.json')
-        : callback(),
+    ({ request }, callback) => {
+      if (/package\.json$/.test(request)) {
+        callback(null, 'commonjs ../../package.json');
+      } else {
+        callback();
+      }
+    },
+
+    // require("node:<package>") に対応していない node バージョンのために、
+    // require("<package>") に変換する
+    ({ request }, callback) => {
+      const module = request.match(/^node:(.+)/)?.[1];
+
+      if (module) {
+        callback(null, `commonjs ${module}`);
+      } else {
+        callback();
+      }
+    },
   ],
 
   resolve: {
@@ -37,12 +52,18 @@ module.exports = {
         loader: 'node-loader',
       },
       {
-        test: /\.tsx?$/,
+        test: /\.ts$/,
         exclude: /node_modules/,
         use: [
           {
             loader: 'esbuild-loader',
-            options: { loader: 'tsx', target: 'node16' },
+            options: {
+              loader: 'ts',
+
+              // >=14.0.0の動作を保証するため、
+              // "node14" ではなく "node14.0.0" を指定する
+              target: 'node14.0.0',
+            },
           },
         ],
       },

